source/manual/errors.html.md

```diff
@@ -8,9 +8,33 @@ last_reviewed_on: 2020-05-18
 review_in: 6 months
 ---
 
-## How we categorise errors
+## How we serve errors on GOV.UK
 
-Sometimes applications will encounter exceptions. This policy describes what we should do for different types of errors. It was first proposed in [RFC 87](https://github.com/alphagov/govuk-rfcs/blob/master/rfc-087-dealing-with-errors.md).
+When a request to GOV.UK fails, we need to handle the error in some way, so that GOV.UK does not look completely broken. Rather than embedding this into every app, we have multiple layers of error handling.
+
+1. **CDN**. When the origin servers [are down](https://github.com/alphagov/govuk-cdn-config/blob/d77313abdb5098e2b350de7a0992375e50ff03a3/vcl_templates/www.vcl.erb#L36), we:
+
+  - Try to [serve the page from a sequence of static mirrors](https://github.com/alphagov/govuk-cdn-config/blob/d77313abdb5098e2b350de7a0992375e50ff03a3/vcl_templates/www.vcl.erb#L274)
+  - Serve [an inline 404 page](https://github.com/alphagov/govuk-cdn-config/blob/d77313abdb5098e2b350de7a0992375e50ff03a3/vcl_templates/www.vcl.erb#L483) if the current mirror returns a 404
+  - Serve [a basic inline error page](https://github.com/alphagov/govuk-cdn-config/blob/d77313abdb5098e2b350de7a0992375e50ff03a3/vcl_templates/www.vcl.erb#L531) if all the mirrors fail (uh oh!)
+
+1. **Origin**. When a request fails in an upstream app, we:
+
+  - [Intercept the response in Nginx](https://github.com/alphagov/govuk-puppet/blob/7dafec7cccd8308ec90c28835de70243d79b323b/modules/router/templates/router_include.conf.erb#L81)
+  - Replace the response with [a pre-rendered one from Static](https://github.com/alphagov/govuk-puppet/blob/7dafec7cccd8308ec90c28835de70243d79b323b/modules/router/manifests/errorpage.pp#L14)
+
+1. **App**. When an app raises an exception, we:
+
+  - Should [catch the error and return an appropriate error code](https://github.com/alphagov/email-alert-frontend/blob/a2bd35b5b17b7da40cd43df9c2756b564597b66e/app/controllers/application_controller.rb#L10)
+  - Return a "500 Internal Server Error" response (by default)
+
```

Comment:
I would also want to signpost to the "How we handle errors in our apps" section here, as otherwise it reads as though our work is done already.

We could probably rewrite the above couple of points as:

```md
1. **App**. When an app raises an exception, we:

  - Should [catch the error and return an appropriate error code](https://github.com/alphagov/email-alert-frontend/blob/a2bd35b5b17b7da40cd43df9c2756b564597b66e/app/controllers/application_controller.rb#L10)
  - We should also log the error so that we can debug it later. See ["How we handle errors in our apps"](#how-we-handle-errors-in-our-apps) below.
```

Reply:
Hopefully the changes in https://github.com/alphagov/govuk-developer-docs/pull/2565#discussion_r428599344 will clarify this better. I'd like to separate the mechanism from the policy; whether we log or not doesn't affect what the user sees.

Does that help?